### PR TITLE
Add new projectusers row when giving project access

### PR DESCRIPTION
**Pull request by i-oden**
## Before submitting this PR

1. **Description:** When making a request to `ProjectAccess` endpoint, the code does not check if the user is listed in the `ProjectUsers` table (if they're a researcher) and instead only adds a new row in `ProjectUserKeys` if one is missing. This means that users can technically have access to projects (because they have a row in `ProjectUserKeys` where the project keys are encrypted and defined) but they are not listed in the project users (because there is no row in `ProjectUsers`) and therefore the API thinks that they do not have access. A feature discussion point is of course if we really need the two separate tables, but for the forseeable future we need to make sure that one is not added without the other at least. This PR adds this fix. 
2. **Jira task / GitHub issue:** DDS-1660
3. **How to test:** _Add information on how someone could manually test this functionality. As detailed as possible._
4. **Type of change:** [_Check the relevant boxes in the section below_](#what-type-of-changes-does-the-pr-contain)
5. **Add docstrings and comments to code**, _even if_ you personally think it's obvious.

## What _type of change(s)_ does the PR contain?

- [ ] New feature
  - [ ] Breaking: _Please describe the reason for the break and how we can fix it._
  - [ ] Non-breaking
- [ ] Database change
  - [ ] Migration _included in PR_
  - [ ] Migration _not needed_
- [x] Bug fix
  - [ ] Breaking: _Please describe the reason for the break and how we can fix it._
  - [x] Non-breaking
- [ ] Security Alert fix
- [ ] Documentation
- [ ] Tests **(only)**
- [ ] Workflow

## Checklist

- [Sprintlog](../SPRINTLOG.md)
  - [x] Added
  - [ ] Not needed (E.g. PR contains _only_ tests)
- Rebase / Update / Merge _from_ base branch (the branch from which the current is forked)
  - [ ] Done
  - [x] Not needed
- Blocking PRs
  - [ ] Merged
  - [x] No blocking PRs
- PR to `master` branch
  - [ ] Yes: Read [the release instructions](https://github.com/ScilifelabDataCentre/dds_web/blob/master/doc/procedures/new_release.md)
    - [ ] I have followed steps 1-8.
  - [x] No

## Actions / Scans

- **Black**: Python code formatter. Does not execute. Only tests.
  Run `black .` locally to execute formatting.
  - [x] Passed
- **Prettier**: General code formatter. Our use case: MD and yaml mainly.
  Run `npx prettier --write .` locally to execute formatting.
  - [x] Passed
- **Yamllint**: Linting of yaml files.
  - [x] Passed
- **Tests**: Pytest to verify that functionality works as expected.
  - [x] New tests added
  - [ ] No new tests
  - [x] Passed
- **CodeQL**: Scan for security vulnerabilities, bugs, errors
  - [ ] New alerts: _Go through them and either fix, dismiss och ignore. Add reasoning in items below._
  - [ ] Alerts fixed: _What?_
  - [ ] Alerts ignored / dismissed: _Why?_
  - [x] Passed
- **Trivy**: Security scanner
  - [ ] New alerts: _Go through them and either fix, dismiss och ignore. Add reasoning in items below._
  - [ ] Alerts fixed: _What?_
  - [ ] Alerts ignored / dismissed: _Why?_
  - [x] Passed
- **Snyk**: Security scanner
  - [ ] New alerts: _Go through them and either fix, dismiss och ignore. Add reasoning in items below._
  - [ ] Alerts fixed: _What?_
  - [ ] Alerts ignored / dismissed: _Why?_
  - [x] Passed

<a href="https://gitpod.io/#https://github.com/ScilifelabDataCentre/dds_web/pull/1464"></a>

### CODE REVIEW

**Comment by rv0lt** [comment]
In _test_fix_access_unitadmin_valid_email_researcher_no_projectuser_row_
The comment says _"""Unit Admin giving access to researcher where"""_
Is that on purpose or are we missing the second part of the sentence ?

**Comment by i-oden** [comment]
> In _test_fix_access_unitadmin_valid_email_researcher_no_projectuser_row_ The comment says _"""Unit Admin giving access to researcher where"""_ Is that on purpose or are we missing the second part of the sentence ?

Not on purpose, no, will fix! Thanks for the comment. 

To make things easier regarding reviews, you can make in-row comments (I'll also add this as documentation, I know I haven't mentioned anything about how we do reviews etc): 
1. Click on "Files changed" 
2. Hover over the line that you want to comment on or suggest a change on; A green plus ➕  will show up on the left side
3. Click on the ➕ 
4. Leave a comment in the "Write field"
5. If you just want to make a comment, then click on "Add single comment" but preferably click on "Start a Review"
6. Leave more comments in a similar way on other rows
7. When you're done, scroll to the top and click "Review Changes" 
8. Leave an (optional) additional comment if you want
9. Click either "Comment" (if you just want to comment), "Approve" (if you approve the PR) or "Request changes" (if you want something to change before you want to approve, like now) 
10. Click "submit review"

image
image
image

File: tests/test_project_access.py
Comment (rv0lt) [inline tests/test_project_access.py:474]: Comments says     _# Fix access for user with no project specified_

But project is specified in _query_string_

File: tests/test_project_access.py
Comment (rv0lt) [inline tests/test_project_access.py:465]: This part of 
 _Delete projectuser row and verify that it's deleted for user_

can be removed as it is encapsulated by the helper function :)

**Review by rv0lt — COMMENTED** [review]
The thing missing is remove this last part as it is done by the function. And I dont see anything else more on my end!

File: tests/test_project_access.py
Comment (i-oden) [reply tests/test_project_access.py:465]: The helper function uses the `ProjectUserKeys` table while this uses the `ProjectUsers` table. 
We could techically add add another argument to the function to specify which table, I'll have a look and see if it simplifies.

Comment (rv0lt) [reply tests/test_project_access.py:465]: ahhh I never saw that difference, sorry. Forget what I said.

**Review by rv0lt — APPROVED** [review]

File: tests/test_project_access.py
Comment (i-oden) [reply tests/test_project_access.py:465]: No worries, very similar!

Comment (i-oden) [reply tests/test_project_access.py]: I added the variable to the helper function now though, I'll tag you as a reviewer again

**Review by rv0lt — APPROVED** [review]

**Review by valyo — APPROVED** [review]
Looks good and I have nothing to add

**Pull request merged**
